bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -2174,6 +2153,131 @@ void ReplicaImp::onCommitVerifyCombinedSigResult(SeqNum seqNumber, ViewNum view,
   startExecution(seqNumber, span, askForMissingInfoAboutCommittedItems);
 }
 
+void ReplicaImp::onFastPathCommitCombinedSigSucceeded(SeqNum seqNumber,
+                                                      ViewNum view,
+                                                      CommitPath cPath,
+                                                      const char *combinedSig,
+                                                      uint16_t combinedSigLen,
+                                                      const concordUtils::SpanContext &span_context) {
+  SCOPED_MDC_PRIMARY(std::to_string(currentPrimary()));
+  SCOPED_MDC_SEQ_NUM(std::to_string(seqNumber));
+  SCOPED_MDC_PATH(CommitPathToMDCString(cPath));
+  LOG_TRACE(THRESHSIGN_LOG, KVLOG(seqNumber, view, combinedSigLen));
+
+  if (isCollectingState() && mainLog->insideActiveWindow(seqNumber)) {
+    mainLog->get(seqNumber).resetCommitSignatures(cPath);
+    LOG_INFO(CNSUS, "Collecting state, reset commit signatures");
+    return;
```

Comment:
Should we delete FullCommitProofMsg for the seqNumber here? It may lead to a mem leak.

Reply:
CollectorOfThresholdSignatures takes ownership of the combinedValidSignatureMsg or candidateCombinedSignatureMsg (the FullCommitProofMsg) depending if it created it or received it for verification from another replica. In both cases CollectorOfThresholdSignatures will delete it either when resetCommitSignatures or the CollectorOfThresholdSignatures is destroyed. The only case we delete a FullCommitProofMsg manually is if we receive it but fail to add it to CollectorOfThresholdSignatures.